File: perl/ase_output_logging.pl
#!/usr/bin/perl
use strict;

# Automation Scripting Essential Skills 
# Perl

# Input: Parsing command-line parameters
# Technique #1 -- Use log4perl 
use Log::Log4perl;

Log::Log4perl::init_and_watch("./log4perl.conf", 10);
our $logger = Log::Log4perl->get_logger('root');

$logger->debug("This is a  debug message.");
$logger->info("This is a  info message.");
$logger->warn("This is a  warn message.");
$logger->error("This is an error message.");
$logger->fatal("This is a  fatal message.");

File: python/ase_output_logging.py
#!/usr/bin/python


# Automation Scripting Essential Skills 
# Python

# Input: Parsing command-line parameters 
# Technique1 -- Use logging 
import logging
import logging.config

LogFileName='./logging.log'
LogLevel='DEBUG'
Verbose=True

def main():

    rval=True

    logging.config.fileConfig('./logging.conf')
    logger = logging.getLogger('root')

    if Verbose:
        # Add the console handler
        ch = logging.StreamHandler()
        ch.setLevel(1)
        formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
        ch.setFormatter(formatter)
        logger.addHandler(ch)


    if rval:
        logger.debug('This is a debug message')
        logger.info('This is an info message')
        logger.warn('This is a warn message')
        logger.error('This is an error message')
        logger.critical('This is a critical message')

if __name__ == "__main__":
    main()
